### PR TITLE
[gsoc] gdbserver #1

### PR DESCRIPTION
**Pull request by SrimantaBarua**
Mostly asking for comments if this is the way to go.
As of now, we can't do much debugging with this.

### CODE REVIEW

File: binr/rules.mk [hunk 1]
```diff
@@ -46,6 +46,7 @@ endif
 endif
 LDFLAGS+=${DL_LIBS}
 LDFLAGS+=${LINK}
+LDFLAGS+=-lr_socket
```
Comment (radare) [inline binr/rules.mk]: why here not all binr/*/ programs will depend on r_socket

**Comment by radare** [comment]
lgtm

**Comment by Maijin** [comment]
fail to build on appveyor and on travis please fix.

**Comment by SrimantaBarua** [comment]
@Maijin fixed

**Pull request merged**